cray_product_catalog/catalog_update.py

```diff
@@ -168,7 +180,16 @@ def update_config_map(data, name, namespace):
 
             # Config map doesn't exist yet
             if e.status == ERR_NOT_FOUND:
```

Comment:
```
# After cm read exception , its reducing nested if/else

# not able to read config map and its a master config map
if e.status == ERR_NOT_FOUND and name == CONFIG_MAP:
    retry read operation 
 else if e.status == ERR_NOT_FOUND :  # product config map not found
    create product 
else:  # retry can be a network issue or something 
  continue 
```
What @lathanm /@abhishek/ @paulose-kuriakose /@anoop, u guys suggest?

Reply:
OR to avoid nesting we can do this (and also avoid multiple checks):
```
if e.status != ERR_NOT_FOUND:
    raise
if name == CONFIG_MAP:
   #If main config map is not found wait until it is available
   LOGGER.warning("ConfigMap %s/%s doesn't exist, attempting again", namespace, name)
else:
   LOGGER.warning("Product ConfigMap %s/%s doesn't exist, attempting to create", namespace, name)
   create_config_map(api_instance, name, namespace)
continue
```

Reply:
@krabhi-hpe , its better just we can make send `if` as `if else`

Reply:
Yes. second one can be `elif`

Reply:
corrected the typo in my comment.